pennylane_lightning/lightning_tensor/_state_tensor.py

```diff
@@ -0,0 +1,187 @@
+        return self._tensor_state
+
+    def _state_dtype(self):
+        """Binding to Lightning Managed state tensor C++ class.
+
+        Returns: the state tensor class
+        """
+        return StateTensorC128 if self.dtype == np.complex128 else StateTensorC64
+
+    def reset_state(self):
+        """Reset the device's state tensor"""
+        # init the state tensor to |00..0>
+        self._tensor_state.reset()
+
+    def _apply_basis_state(self, state, wires):
+        """Initialize the state tensor in a specified computational basis state.
+
+        Args:
+            state (array[int]): computational basis state of shape ``(wires,)``
+                consisting of 0s and 1s.
+            wires (Wires): wires that the provided computational state should be
+                initialized on
+
+        Note: This function does not support broadcasted inputs yet.
+        """
+        # length of basis state parameter
+        n_basis_state = len(state)
+
+        if not set(state.tolist()).issubset({0, 1}):
+            raise ValueError("BasisState parameter must consist of 0 or 1 integers.")
+
+        if n_basis_state != len(wires):
+            raise ValueError("BasisState parameter and wires must be of equal length.")
+
+        self._tensor_state.setBasisState(state)
+
+    def _apply_lightning(self, operations):
+        """Apply a list of operations to the state tensor.
+
+        Args:
+            operations (list[~pennylane.operation.Operation]): operations to apply
+
+        Returns:
+            None
+        """
+        state = self.state_tensor
+
+        # Skip over identity operations instead of performing
+        # matrix multiplication with it.
+        for operation in operations:
+            if isinstance(operation, qml.Identity):
+                continue
+            if isinstance(operation, Adjoint):
+                raise DeviceError(
+                    "Adjoint operations are not supported."
+                )  # inverse=False is not supported
+            name = operation.name
+            method = getattr(state, name, None)
+            wires = list(operation.wires)
+
+            if method is not None:  # apply specialized gate
+                param = operation.parameters
+                method(wires, False, param)
+            else:  # apply gate as a matrix
+                # Inverse can be set to False since qml.matrix(operation) is already in
+                # inverted form
+                method = getattr(state, "applyMatrix")
+                try:
+                    method(qml.matrix(operation), wires, False)
+                except AttributeError:  # pragma: no cover
+                    # To support older versions of PL
+                    method(operation.matrix, wires, False)
+
+    def apply_operations(self, operations):
+        """Append operations to underly graph."""
+        # State preparation is currently done in Python
+        if operations:  # make sure operations[0] exists
```

Comment:
would you like to raise a device error for ops with num_wires > 2 here?

Reply:
It's already defined by the `stopping_condition_mps` method in `lightning_tensor.py`